ops/model.py

```diff
@@ -230,6 +236,81 @@ def _get_unique(self, relation_name, relation_id=None):
             raise TooManyRelatedAppsError(relation_name, num_related, 1)
 
 
+class BindingMapping:
+    """A map of binding names to lists of Binding instances."""
+
+    def __init__(self, backend):
+        self._backend = backend
+        self._data = {}
+
+    def get(self, relation):
+        if not isinstance(relation, Relation):
+            raise ModelError(f'a relation argument passed is not a Relation: {type(relation).__name__}')
+        binding = self._data.get(relation)
+        if binding is None:
+            self._data[relation] = binding = Binding(relation.name, relation.id, self._backend)
+        return binding
+
+
+class Binding:
+    """A representation of a binding to a space."""
+
+    def __init__(self, name, relation_id, backend):
+        self.name = name
+        if not isinstance(relation_id, int):
+            raise ModelError(f'relation_id must be an int, not {type(relation_id).__name__}')
+        self._relation_id = relation_id
+        self._backend = backend
+        self._network = None
+
+    @property
+    def network(self):
+        if self._network is None:
+            self._network = Network(self._backend.network_get(self.name, self._relation_id))
+        return self._network
+
+
+class Network:
+    """A representation of a unit's view of the network associated with a network space binding."""
```

Comment:
```
"""Network space details."""
```
This is the network space itself. The `binding.network` is what refers to it, not the opposite.